manifests/daemon.pp

```diff
@@ -79,6 +82,7 @@
   Boolean $purge                       = true,
   String $options                      = '',
   String $init_style                   = $prometheus::init_style,
+  Boolean $deploy_unit_file            = true,
```

Comment:
what about making this a generic parameter like `manage_service_file`? Maybe people also want to use their custom sysv file.